### ISSUE TITLE
systemProbe.enableTCPQueueLength causing golang panics

### ISSUE THREAD

**Issue by kr3cj** [DataDog/datadog-agent#7180]
After enabling the TCP Queue Length network monitoring on our datadog agent systemprobe container, it restarts a lot (`Last State: Terminated`), logs golang panics, and no TCP Queue metrics are generated. By one measure, it was happening about 27 times per minute from one node.

```sh
2021/01/13 16:26:41 http: panic serving @: reflect: Field index out of range
goroutine 60 [running]:
net/http.(*conn).serve.func1(0xc0002ebe00)
/root/.gimme/versions/go1.14.7.linux.amd64/src/net/http/server.go:1800 +0x139
"panic(0x1fd03e0, 0x28e0e10)"
/root/.gimme/versions/go1.14.7.linux.amd64/src/runtime/panic.go:975 +0x3e3
encoding/json.(*encodeState).marshal.func1(0xc000247a80)
/root/.gimme/versions/go1.14.7.linux.amd64/src/encoding/json/encode.go:326 +0x9a
"panic(0x1fd03e0, 0x28e0e10)"
/root/.gimme/versions/go1.14.7.linux.amd64/src/runtime/panic.go:969 +0x166
"reflect.Value.Field(0x21e7280, 0xc0018e21e8, 0x99, 0x377000000ca, 0x41a99f, 0x21e7280, 0xc0018e21e8)"
/root/.gimme/versions/go1.14.7.linux.amd64/src/reflect/value.go:838 +0xd9
"encoding/json.structEncoder.encode(0xc00001d0e0, 0x2, 0x2, 0xc0015bd8c0, 0xc000632600, 0x21e7280, 0xc0018e21e8, 0x99, 0x1f70100)"
/root/.gimme/versions/go1.14.7.linux.amd64/src/encoding/json/encode.go:745 +0x81
"encoding/json.mapEncoder.encode(0xc0015bd8f0, 0xc000632600, 0x213df20, 0xc00168c540, 0x15, 0x2130100)"
/root/.gimme/versions/go1.14.7.linux.amd64/src/encoding/json/encode.go:801 +0x354
"encoding/json.(*encodeState).reflectValue(0xc000632600, 0x213df20, 0xc00168c540, 0x15, 0xc000240100)"
/root/.gimme/versions/go1.14.7.linux.amd64/src/encoding/json/encode.go:358 +0x82
"encoding/json.(*encodeState).marshal(0xc000632600, 0x213df20, 0xc00168c540, 0xc000240100, 0x0, 0x0)"
/root/.gimme/versions/go1.14.7.linux.amd64/src/encoding/json/encode.go:330 +0xf0
"encoding/json.Marshal(0x213df20, 0xc00168c540, 0x8, 0xc000247b20, 0x1cc8687, 0xc000289580, 0x0)"
/root/.gimme/versions/go1.14.7.linux.amd64/src/encoding/json/encode.go:161 +0x52
"github.com/DataDog/datadog-agent/cmd/system-probe/utils.WriteAsJSON(0x2982aa0, 0xc00155a2a0, 0x213df20, 0xc00168c540)"
/go/src/github.com/DataDog/datadog-agent/cmd/system-probe/utils/utils.go:12 +0x39
"github.com/DataDog/datadog-agent/cmd/system-probe/modules.(*tcpQueueLengthModule).Register.func1(0x2982aa0, 0xc00155a2a0, 0xc00163a100)"
/go/src/github.com/DataDog/datadog-agent/cmd/system-probe/modules/tcp_queue_tracer.go:41 +0x5b
"net/http.HandlerFunc.ServeHTTP(0xc000289620, 0x2982aa0, 0xc00155a2a0, 0xc00163a100)"
/root/.gimme/versions/go1.14.7.linux.amd64/src/net/http/server.go:2041 +0x44
"net/http.(*ServeMux).ServeHTTP(0xc00014f7c0, 0x2982aa0, 0xc00155a2a0, 0xc00163a100)"
/root/.gimme/versions/go1.14.7.linux.amd64/src/net/http/server.go:2416 +0x1a5
"net/http.serverHandler.ServeHTTP(0xc00017c0e0, 0x2982aa0, 0xc00155a2a0, 0xc00163a100)"
/root/.gimme/versions/go1.14.7.linux.amd64/src/net/http/server.go:2836 +0xa3
"net/http.(*conn).serve(0xc0002ebe00, 0x298b0a0, 0xc000062300)"
/root/.gimme/versions/go1.14.7.linux.amd64/src/net/http/server.go:1924 +0x86c
created by net/http.(*Server).Serve
/root/.gimme/versions/go1.14.7.linux.amd64/src/net/http/server.go:2962 +0x35c
```

Which seems to trace back to [this line](https://github.com/DataDog/datadog-agent/blob/master/cmd/system-probe/modules/tcp_queue_tracer.go#L43). Somewhat relatedly, we also saw many "Too Many CPUs" log messages from [this line](https://github.com/DataDog/datadog-agent/blob/1ee0bc9aceba46b0968b1412e771ae7e0890169e/pkg/collector/corechecks/ebpf/probe/tcp_queue_length.go#L112) as well (even though our CPU resources are lower; see below).

**Describe what you expected:**
TCP Queue metrics to show up in DataDog such as `system.net.tcp.recv_q.avg` without an impact to the system-probe container health or stability.

**Steps to reproduce the issue:**
Enable TCPQueueLength on the datadog agent. We are installing and configuring the datadog agent (v7.24.1) with dedicated containers via the helm chart (v2.6.7):

```sh
datadog:
...
  systemProbe:
    enableOOMKill: true
    collectDNSStats: true
    enableTCPQueueLength: true
..
agents:
  enabled: true
...
  containers:
...
    systemProbe:
      resources:
        requests:
          cpu: 125m
          memory: 256Mi
        limits:
          cpu: 1
          memory: 256Mi
```

**Additional environment details (Operating System, Cloud provider, etc):**
We are running on EKS 1.18. We are using the standard EKS AMI amazon linux image ([latest version](https://github.com/awslabs/amazon-eks-ami/releases/tag/v20201211) at the time of writing). In order to get eBPF working, we had to install extra packages via the eksctl's `preBootstrapCommands`:

```sh
    preBootstrapCommands:
      # datadog agent needs these to get ebpf working to grab oom_kill and tcp queue length metrics
      - sudo yum install kernel-headers-$(uname --kernel-release) -y
      # avoid installing 60 dependencies and increasing bootstrap time by 90 seconds
      - sudo rpm -Uvh --nodeps $(repoquery --location kernel-devel-$(uname --kernel-release))
```

```sh
# commands run on the node, not the container
$ uname --kernel-release
4.14.209-160.335.amzn2.x86_64
$ sudo getenforce
Disabled
```

**Comment by L3n41c** [comment]
This panic should be fixed by #7536 which will be included in agent `7.27.0`.

**Comment by kr3cj** [comment]
Thank you @L3n41c
